### PR TITLE
[cctag] update to v1.0.4

### PR DESCRIPTION
**Pull request by simogasp**
It updates cctag to the new 1.0.4 version.
No patch is required now as the issues with the new boost version 1.85 have been integrated.
Incidentally, it removes the line that removes the findTbb.cmake as it is not shipped anymore.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
@simogasp, Thanks for your PR, when I tested the feature by command `./vcpkg install cctag[*]:x64-windows` I get this error.
Please take a look:
````
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include\crt/host_config.h(153): fatal error C1189: #error:  -- unsupported Microsoft Visual Studio version! Only the versions between 2017 and 2022 (inclusive) are supported! The nvcc flag '-allow-unsupported-compiler' can be used to override this version check; however, using an unsupported host compiler may cause compilation failure or incorrect run time execution. Use at your own risk.

nvcc warning : The 'compute_35', 'compute_37', 'sm_35', and 'sm_37' architectures are deprecated, and may be removed in a future release (Use -Wno-deprecated-gpu-targets to suppress warning).

CMake Error at CCTag_generated_assist.cu.obj.Debug.cmake:220 (message):
  Error generating
  F:/Feature-test/cctag/buildtrees/cctag/x64-windows-dbg/src/CMakeFiles/CCTag.dir/./cctag/cuda/./CCTag_generated_assist.cu.obj
````
The VS I used for local testing is 2022, but it doesn't recognize .

**Comment by simogasp** [comment]
I will try later when I have access to my Windows machine, but I suspect
that you have a too recent version of VS that it is not supported by that
version of cuda. Are you able to compile other packages that uses cuda?
(E.g. popsift)

**Comment by simogasp** [comment]
> @simogasp, Thanks for your PR, when I tested the feature by command `./vcpkg install cctag[*]:x64-windows` I get this error. Please take a look:
> 
> ```
> C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include\crt/host_config.h(153): fatal error C1189: #error:  -- unsupported Microsoft Visual Studio version! Only the versions between 2017 and 2022 (inclusive) are supported! The nvcc flag '-allow-unsupported-compiler' can be used to override this version check; however, using an unsupported host compiler may cause compilation failure or incorrect run time execution. Use at your own risk.
> 
> nvcc warning : The 'compute_35', 'compute_37', 'sm_35', and 'sm_37' architectures are deprecated, and may be removed in a future release (Use -Wno-deprecated-gpu-targets to suppress warning).
> 
> CMake Error at CCTag_generated_assist.cu.obj.Debug.cmake:220 (message):
>   Error generating
>   F:/Feature-test/cctag/buildtrees/cctag/x64-windows-dbg/src/CMakeFiles/CCTag.dir/./cctag/cuda/./CCTag_generated_assist.cu.obj
> ```
> 
> The VS I used for local testing is 2022, but it doesn't recognize .

can you check the conditions around line 153 of `C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include\crt/host_config.h`?

I should be something 
```c++
#if defined(_WIN32)

#if _MSC_VER < 1910 || _MSC_VER >= 1930

#error -- unsupported Microsoft Visual Studio version! Only the versions between 2017 and 2019 (inclusive) are supported! The nvcc flag '-allow-unsupported-compiler' can be used to override this version check; however, using an unsupported host compiler may cause compilation failure or incorrect run time execution. Use at your own risk.

```
this is mine for cuda 11.5.
Can you check the value `_MSC_VER >= XXXX` is in your version of cuda and which exact version of VS you are using?

**Comment by BillyONeal** [comment]
@JonLiu1993 CI isn't detecting this probably because it uses CUDA 12.1 rather than 11.7 as in your example. You probably need to update your copy of CUDA to a 12.x release for VS2022.

https://github.com/microsoft/vcpkg/blob/89e222a90fc2847600be430a85783b8943f31f6c/scripts/azure-pipelines/windows/deploy-cuda.ps1 from https://github.com/microsoft/vcpkg/pull/39325 will install CUDA 12.4 for you which might help

**Comment by JonLiu1993** [comment]
Thanks @BillyONeal @simogasp, I updated my local computer cuda to 12.5 and successfully tested the feature.
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Review by JonLiu1993 — APPROVED** [review]

File: ports/cctag/portfile.cmake [hunk 1]
```diff
@@ -2,14 +2,10 @@ vcpkg_from_github(
...
     REPO alicevision/cctag
     REF "v${VERSION}"
-    SHA512 ccae0829480eca2a25cae70a5bf2cb3fde5ae1f71a08a9b614c05a1bb746ad66d6ac70f590ac8096635ea4fe75590d0ecb1c8c9f1b77189e5eae6f5e59c78cfe
+    SHA512 65f8260213faab1965ff97e38b890d85f5599c5db5f50f09ab1ed2d73d7008d2e93693145d66a4d9af6342666817204736c3b0384885fa50402850f1dc5dceae
     HEAD_REF develop
-    PATCHES
-        fix-compatibility-with-boost-1.85.patch
 )
 
-file(REMOVE "${SOURCE_PATH}/cmake/FindTBB.cmake" "${SOURCE_PATH}/src/applications/cmake/FindTBB.cmake")
```
Comment (BillyONeal) [inline ports/cctag/portfile.cmake:11]: Obsoleted by https://github.com/alicevision/CCTag/commit/95bbb11312e741f1b8d1f76022d8849aeb9546a2

File: ports/cctag/portfile.cmake [hunk 2]
```diff
@@ -2,14 +2,10 @@ vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO alicevision/cctag
     REF "v${VERSION}"
-    SHA512 ccae0829480eca2a25cae70a5bf2cb3fde5ae1f71a08a9b614c05a1bb746ad66d6ac70f590ac8096635ea4fe75590d0ecb1c8c9f1b77189e5eae6f5e59c78cfe
+    SHA512 65f8260213faab1965ff97e38b890d85f5599c5db5f50f09ab1ed2d73d7008d2e93693145d66a4d9af6342666817204736c3b0384885fa50402850f1dc5dceae
     HEAD_REF develop
-    PATCHES
-        fix-compatibility-with-boost-1.85.patch
```
Comment (BillyONeal) [inline ports/cctag/portfile.cmake:8]: Merged as https://github.com/alicevision/CCTag/pull/218

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**